### PR TITLE
Elsi sparse review

### PR DESCRIPTION
**Pull request by bhourahine**

### CODE REVIEW

File: prog/dftb+/lib_elecsolvers/elecsolvers.F90 [hunk 1]
```diff
@@ -47,6 +47,19 @@ module dftbp_elecsolvers
...
+
+    !> Whether the solver provides the TS term for electrons
+    logical, public :: providesElectronEntropy
+
+    !> Whether the solver provides electronic free energy (or is consistent with its evaluation)
+    logical, public :: providesFreeEnergy
+
+    !> Whether the solver provides the electron chemical potential (or is consistent with its
+    !> evaluation or it being suplied externally)
+    logical, public :: elecChemPotAvailable
```
Comment (aradi) [inline prog/dftb+/lib_elecsolvers/elecsolvers.F90:61]: Why not `providesElecChemPot`?

File: prog/dftb+/lib_elecsolvers/elsisolver.F90 [hunk 2]
```diff
@@ -490,9 +490,8 @@ subroutine supportedVersionNumber(major, minor, patch, dateStamp)
       isSupported = .false.
     elseif (major == 2 .and. minor < 6) then
       isSupported = .false.
-    elseif (major == 2 .and. minor == 6 .and. patch < 0) then
-      ! no need to check patch in this case, as library must be >= 2.6.0, but just in case of a
-      ! negative value
+    elseif (major == 2 .and. minor == 6 .and. all(patch == [0,1])) then
+      ! library must be 2.6.{0,1}
```
Comment (aradi) [inline prog/dftb+/lib_elecsolvers/elsisolver.F90:494]: `all(patch == [0, 1])` will be always `.false.`. Why don't we leave this away and keep only the checks ensuring not being older than 2.6.0?

**Review by aradi — APPROVED** [review]
:+1:

File: prog/dftb+/lib_elecsolvers/elecsolvers.F90 (hunk 1 above)
Comment (bhourahine) [reply prog/dftb+/lib_elecsolvers/elecsolvers.F90:61]: That name would be slightly misleading, as when its true the chemical potential may either be
1. actually provided by the solver (libOMM, NTPoly, PEXSI, ...)
2. specified by using the solver output combined with the required number of electrons (eigenvalues returned and we work out the fillings)
3. specified by an external bath/contact and then required by the solver (NEGF, ...)

File: prog/dftb+/lib_elecsolvers/elsisolver.F90 (hunk 2 above)
Comment (bhourahine) [reply prog/dftb+/lib_elecsolvers/elsisolver.F90:494]: Replaced with any() instead of all()

**Pull request merged**